substrate/client/sysinfo/src/sysinfo.rs

```diff
@@ -375,6 +379,49 @@ pub fn benchmark_cpu(limit: ExecutionLimit) -> Throughput {
 		.expect("benchmark cannot fail; qed")
 }
 
+// This benchmarks the entire CPU speed as measured by calculating BLAKE2b-256 hashes, in bytes per
+// second. It spawns multiple threads to measure the throughput of the entire CPU and averages the
+// score obtained by each thread. If we have at least `EXPECTED_NUM_CORES` available then the
+// average throughput should be relatively close to the single core performance as measured by
+// `benchmark_cpu`.
+pub fn benchmark_cpu_parallelism(limit: ExecutionLimit) -> Throughput {
+	const SIZE: usize = 32 * 1024;
+	const EXPECTED_NUM_CORES: usize = 8;
```

Comment:
Seems to be going in the right direction, but his is a *substrate* crate - not Polkadot. There may be non-Polkadot chains that have different requirements. This is why all the requirements are passed into this crate [here](https://github.com/paritytech/polkadot-sdk/blob/f1c16d05e77fc1fc3a98ae4e21793edcb4d301df/substrate/utils/frame/benchmarking-cli/src/machine/hardware.rs#L28).  
This `SUBSTRATE_REFERENCE_HARDWARE` is then used by the Polkadot and parachain nodes, but maybe not by other projects.  
I think the easiest way would be to also add a `metric` here with `NumCpu`: https://github.com/paritytech/polkadot-sdk/blob/7ecf3f757a5d6f622309cea7f788e8a547a5dce8/substrate/client/sysinfo/src/sysinfo.rs#L40

Reply:
I'm not convinced that the number of threads should be configurable here, simply because that will make the same benchmark produce different numbers depending on the number of threads picked here.

If we want to make the number of threads configurable then the number of threads should be included in the benchmark's name so that there's no potential for confusion.

Reply:
My take on @ggwpez proposal is that.
- The enum variant becomes
```
Blake2256Parallel { num_cores: usize }
```

- The serialization in reference_hardware.json becomes:
```
"metric": {"Blake2256Parallel":{"num_cores":8}},
or
"metric": {"Blake2256Parallel":{"num_cores":4}},
```
- The benchamark will then be named, like this when you run `Running machine benchmarks`

```
| CPU      | BLAKE2-256-Parallel-8 | 1.46 GiBs   | 783.27 MiBs | ✅ Pass (190.4 %) |
or
| CPU      | BLAKE2-256-Parallel-4 | 1.47 GiBs   | 783.27 MiBs | ✅ Pass (192.5 %) |
```

The advantage here being, that you can allow people to configure their own parallel benchmarks without having to modify the substrate code.

What do you think @koute.

Reply:
As long as the benchmarks have different names so that there is no confusion I'm fine with it.

Reply:
>Blake2256Parallel { num_cores: usize }

Yep thats an even better idea!

Reply:
Ok, turns out we need 2 changes to the format to keep this general and not enforce the checking of the parallel benchmark on non-relay chain authorities and parachains:
-  the enum variant becomes:
```
Blake2256Parallel { num_cores: usize }
```
- and the Metric gets a new field, telling us if it needs to be checked on authorithies or not 

See: https://github.com/paritytech/polkadot-sdk/pull/5127/files#diff-633d5244ab57ee3488db02991063e8f4562004df4200a04e251d0caea145018eR6.

Let me know what you think.

The alternative would be to have a different reference_hardware.json for polkadot binary, but that will easily get out of sync with the substrate/reference_hardware.json, so I tried to avoid it.